### ISSUE TITLE
[Sitemaps] Disable XML resolvers

### ISSUE THREAD

**Issue by sebastian-nagel** [crawler-commons/crawler-commons#151]
Sometimes processing a sitemap (including XHTML documents delivered as content when a sitemap.xml is requested) takes long (minutes), ev. causing Hadoop MapReduce tasks to fail with a timeout. These processes hang with a connection to hans-moleman.w3.org (cf. [[1](http://strangersgeospatial.blogspot.de/2013/05/hans-moleman-issues.html)])
```
java    6823 yarn  390u  IPv4             490354      0t0      TCP ip-10-143-152-145.ec2.internal:33223->hans-moleman.w3.org:http (CLOSE_WAIT)
java    6823 yarn  391u  IPv4             505696      0t0      TCP ip-10-143-152-145.ec2.internal:33228->hans-moleman.w3.org:http (ESTABLISHED)
```
opened by XMLEntityManager:
```
        at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1325)
        - locked <0x000000074fbc0430> (a sun.net.www.protocol.http.HttpURLConnection)
        at org.apache.xerces.impl.XMLEntityManager.setupCurrentEntity(Unknown Source)
```

To process sitemaps, there should be no need for these remote resources. The problem affects both DOM and SAX parser.

**Comment by kkrugler** [comment]
Sounds like we need to register a custom entity resolver. I know that's possible with the SAX parser, I assume it's the same for DOM. There was this interesting [post](http://strangersgeospatial.blogspot.com/2013/05/hans-moleman-issues.html) about the same issue.

**Comment by kkrugler** [comment]
Hmm, looks like Xerces might have a [resolver](http://xerces.apache.org/xml-commons/components/resolver/) to handle this situation. I've found snippets re using this with Dom4J and SAX parsers.

**Comment by kkrugler** [comment]
Or you could just disable validation :)

**Comment by sebastian-nagel** [comment]
Yes, a custom noop entity resolver or disabling validation resolves the problem, see #152.
